arduino/sketch/profiles.go

```diff
@@ -0,0 +1,213 @@
+	Libraries ProfileRequiredLibraries `yaml:"libraries"`
+}
+
+// AsYaml outputs the profile as Yaml
+func (p *Profile) AsYaml() string {
+	res := ""
+	if p.Notes != "" {
+		res += fmt.Sprintf("    notes: %s\n", p.Notes)
+	}
+	res += fmt.Sprintf("    fqbn: %s\n", p.FQBN)
+	res += p.Platforms.AsYaml()
+	res += p.Libraries.AsYaml()
+	return res
+}
+
+// ProfileRequiredPlatforms is a list of ProfilePlatformReference (platforms
+// required to build the sketch using this profile)
+type ProfileRequiredPlatforms []*ProfilePlatformReference
+
+// AsYaml outputs the required platforms as Yaml
+func (p *ProfileRequiredPlatforms) AsYaml() string {
+	res := ""
+	for _, platform := range *p {
+		res += platform.AsYaml()
+	}
+	return res
+}
+
+// ProfileRequiredLibraries is a list of ProfileLibraryReference (libraries
+// required to build the sketch using this profile)
+type ProfileRequiredLibraries []*ProfileLibraryReference
+
+// AsYaml outputs the required libraries as Yaml
+func (p *ProfileRequiredLibraries) AsYaml() string {
+	return ""
+}
+
+// ProfilePlatformReference is a reference to a platform
+type ProfilePlatformReference struct {
+	Packager         string
+	Architecture     string
+	Version          *semver.Version
+	PlatformIndexURL *url.URL
+}
+
+// InternalUniqueIdentifier returns the unique identifier for this object
+func (p *ProfilePlatformReference) InternalUniqueIdentifier() string {
+	id := p.String()
+	h := sha256.Sum256([]byte(id))
+	res := fmt.Sprintf("%s:%s@%s_%s", p.Packager, p.Architecture, p.Version, hex.EncodeToString(h[:])[:16])
+	return utils.SanitizeName(res)
+}
+
+func (p *ProfilePlatformReference) String() string {
+	res := fmt.Sprintf("%s:%s@%s", p.Packager, p.Architecture, p.Version)
+	if p.PlatformIndexURL != nil {
+		res += fmt.Sprintf(" (%s)", p.PlatformIndexURL)
+	}
+	return res
+}
+
+// AsYaml outputs the platform reference as Yaml
+func (p *ProfilePlatformReference) AsYaml() string {
+	res := fmt.Sprintf("      - platform: %s:%s (%s)\n", p.Packager, p.Architecture, p.Version)
+	if p.PlatformIndexURL != nil {
+		res += fmt.Sprintf("        platform_index_url: %s\n", p.PlatformIndexURL)
+	}
+	return res
+}
+
+func parseNameAndVersion(in string) (string, string, bool) {
+	re := regexp.MustCompile(`^([a-zA-Z0-9.\-_ :]+) \((.+)\)$`)
+	split := re.FindAllStringSubmatch(in, -1)
+	if len(split) != 1 || len(split[0]) != 3 {
+		return "", "", false
+	}
+	return split[0][1], split[0][2], true
+}
+
+// UnmarshalYAML decodes a ProfilePlatformReference from YAML source.
+func (p *ProfilePlatformReference) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var data map[string]string
+	if err := unmarshal(&data); err != nil {
+		return err
+	}
+	if platformID, ok := data["platform"]; !ok {
+		return fmt.Errorf(tr("missing 'platform' directive"))
+	} else if platformID, platformVersion, ok := parseNameAndVersion(platformID); !ok {
+		return fmt.Errorf(tr("invalid 'platform' directive"))
```

Comment:
```suggestion
		return fmt.Errorf(tr("missing '%s' directive", "platform"))
	} else if platformID, platformVersion, ok := parseNameAndVersion(platformID); !ok {
		return fmt.Errorf(tr("invalid '%s' directive", "platform"))
```
`platform` is a key name and so it should not be localized.